src/e2e-test/java/io/cdap/plugin/bigquery/stepsdesign/BigQueryMultiTableValidation.java

```diff
@@ -0,0 +1,109 @@
+import java.util.stream.Collectors;
+
+/**
+ * BQValidation.
+ */
+public class BigQueryMultiTableValidation {
+  static Gson gson = new Gson();
+  public static boolean validateBQToBigQueryMultiTable(List<String> sourceTables, List<String> targetTables)
+    throws IOException, InterruptedException {
+    // Check if there's only one source table
+    if (sourceTables.size() == 1) {
+      String currentTargetTable = targetTables.get(0);
+      return validateBQToBQSingleTable(sourceTables.get(0), currentTargetTable);
+    } else {
+      if (sourceTables.size() != targetTables.size()) {
+        throw new IllegalArgumentException("Number of source tables and target tables must be the same.");
+      }
+      // Iterate over each pair of source and target tables
+      for (int i = 0; i < sourceTables.size(); i++) {
+        String currentSourceTable = sourceTables.get(i);
+        String currentTargetTable = targetTables.get(i);
+        // Perform validation for the current pair of tables
+        if (!validateBQToBQSingleTable(currentSourceTable, currentTargetTable)) {
+          return false; // Return false if validation fails for any table pair
+        }
+      }
+      // Return true if validation passes for all table pairs
+      return true;
+    }
+  }
+
+  public static boolean validateBQToBQSingleTable(String sourceTable, String targetTable)
+    throws IOException, InterruptedException {
+    // Fetch data from the source BigQuery table
+    List<Object> bigQueryRows = new ArrayList<>();
+    getBigQueryTableData(sourceTable, bigQueryRows);
+
+    // Convert fetched data into JSON objects
+    List<JsonObject> bigQueryResponse = new ArrayList<>();
+    for (Object row : bigQueryRows) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(row), JsonObject.class);
+      bigQueryResponse.add(jsonData);
+    }
+    // Fetch data from the target BigQuery table
+    List<Object> bigQueryRows2 = new ArrayList<>();
+    getBigQueryTableData(targetTable, bigQueryRows2);
+    List<JsonObject> bigQueryResponse2 = new ArrayList<>();
+    for (Object row : bigQueryRows2) {
+      JsonObject jsonData = gson.fromJson(String.valueOf(row), JsonObject.class);
+      bigQueryResponse2.add(jsonData);
+    }
+    return compareBigQueryDataAndBQMT(bigQueryResponse, bigQueryResponse2);
+  }
+
+  private static void getBigQueryTableData(String table, List<Object> bigQueryRows) throws IOException,
+    InterruptedException {
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + table + "` AS t";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+    result.iterateAll().forEach(value -> bigQueryRows.add(value.get(0).getValue()));
+  }
+
+  private static boolean compareBigQueryDataAndBQMT(List<JsonObject> bigQueryResponse, List<JsonObject> bqmtData)
```

Comment:
A better way to compare is to write concrete classes and use equals method.

Reply:
same comment as above.